src/ansiblelint/rules/pattern.py

```diff
@@ -0,0 +1,121 @@
+"""Implementation of PatternRule."""
+
+from __future__ import annotations
+
+import sys
+from typing import TYPE_CHECKING, Any
+
+from ansiblelint.rules import AnsibleLintRule
+
+if TYPE_CHECKING:
+    from ansiblelint.errors import MatchError
+    from ansiblelint.file_utils import Lintable
+
+
+class PatternRule(AnsibleLintRule):
+    """Rule for checking pattern directory."""
+
+    id = "pattern"
+    description = "Confirm that pattern has valid directory structure."
+    severity = "MEDIUM"
+    tags = ["metadata"]
+    version_changed = "25.7.0"
+
+    def matchplay(self, file: Lintable, data: dict[str, Any]) -> list[MatchError]:
+        """Return matches found for a specific play (entry in playbook)."""
+        if file.kind != "pattern":
+            return []
+
+        results = []
+
+        pattern_dir = file.path.parent.parent.resolve()
+        meta_dir = pattern_dir / "meta"
+
+        # Check if meta directory exists
+        if not meta_dir.is_dir():
+            results.append(
+                self.create_matcherror(
+                    message=(
+                        f"Pattern directory '{pattern_dir}' contains pattern.json but is missing the required 'meta' directory."
+                    ),
+                    tag=self.id,
```

Comment:
Use subtags to distinguish between various sub-rules and document all of them in the readme. `tag=f"{self.id}[xxx]",`.